awx/main/models/inventory.py

```diff
@@ -2616,6 +2616,9 @@ def build_script_private_data(self, inventory_update, private_data_dir):
         group_patterns = '[]'
         group_prefix = 'foreman_'
         want_hostcollections = 'False'
+        want_ansible_ssh_host = 'False'
+        rich_params = 'False'
```

Comment:
This looks to me like it's going to be trouble

https://github.com/ansible/awx/blob/30688ee28ba7772c14f1b5569e018e81cbbb5e4c/awx/plugins/inventory/foreman.ini.example#L181-L186

So... if this is false, we're talking about hostvars not having nested data structures, but having the JSON string-ified versions of them?

Moving to the plugin, I would assume, hope, that the nested data structures are used by default, so maybe having this param set true would be more congruent. That may be an argument for adding it, but it may be impossible to provide backwards support.

Reply:
https://github.com/ansible/ansible/blob/devel/contrib/inventory/foreman.py#L186-L190

Checked on the script and confirmed `rich_params` is False by default. 

> Moving to the plugin, I would assume, hope, that the nested data structures are used by default, so maybe having this param set true would be more congruent. That may be an argument for adding it, but it may be impossible to provide backwards support.

@AlanCoding - aren't we aiming for as much parity as possible between the scripts and plug-ins? (even if a change in behavior would seem like an improvement, thought that parity trumps all)

Reply:
If we're aiming for parity, then this change looks okay b/c it reflects what's in the script.